app/models/user_column.rb

```diff
@@ -0,0 +1,12 @@
+class UserColumn < ApplicationRecord
+  belongs_to :user
+
+  validates :user_id, :resource, :presence => true
+
+  serialize :columns
+
+  # Returns user columns representation as the hash object used in memory
+  def to_hash
+    { :columns => columns, :resource => resource }
```

Comment:
attr_exportable might be usable here instead, take a look at the exportable concern